### ISSUE TITLE
post_train_grpo.sh extremely long training time on 2×4090 GPUs

### ISSUE THREAD

**Issue by ShinPei34520** [shangshang-wang/Tina#6]
When I run post_train_grpo.sh on two NVIDIA RTX 4090 GPUs, the estimated remaining training time is orders of magnitude longer than expected. I have verified that the environment is set up correctly and that the evaluation scripts (post_train_eval_baselines.sh and post_train_eval_local.sh, using the publicly released Hugging Face checkpoint) both run without issues.

I followed the README to configure the environment exactly. In post_train_grpo.sh, the dataset used is "curated_deepscaler". In model_curated_deepscaler.yaml, I applied two changes to avoid OOM:
``` yaml
gradient_accumulation_steps: 8    # originally 4
per_device_train_batch_size: 2    # originally 4
```
After ~24 hours of training, the log reports Remaining time: 463 hours

![Image](https://github.com/user-attachments/assets/3602c323-b6a5-4c17-8517-4fca63fcc0f9)

But Official Tina-DeepScaleR training logs report a total duration of 2 days, 4 hours, 1 minute

![Image](https://github.com/user-attachments/assets/301183d8-1ed5-4ba7-9f74-74fd670f39e7)

Questions & Requests

Could I be missing any performance-related options in model_curated_deepscaler.yaml?

Are there known GRPO bottlenecks on the 4090 architecture?

What batch size / accumulation step / memory settings would you recommend for 2×4090?

Should I adjust gradient synchronization or distributed training settings in the script?

Thank you for your help!

**Comment by shangshang-wang** [comment]
Hey thanks for asking. For your questions:

> Could I be missing any performance-related options in model_curated_deepscaler.yaml?

No, I don't think so

> Are there known GRPO bottlenecks on the 4090 architecture?

Probably not

> What batch size / accumulation step / memory settings would you recommend for 2×4090?

I would say keep the accumulation step fixed and only tune the batch size

> Should I adjust gradient synchronization or distributed training settings in the script?

If you want to modify deeper configs, you might want to learn more about deepspeed, accelerate and other techniques used in trl
